ur_robot_driver/test/robot_driver.py

```diff
@@ -126,22 +148,48 @@ def generate_test_description(tf_prefix):
         ],
         name="start_ursim",
         output="screen",
+        condition=IfCondition(launch_ursim),
     )
+
     wait_dashboard_server = ExecuteProcess(
         cmd=[
             PathJoinSubstitution(
                 [FindPackagePrefix("ur_robot_driver"), "bin", "wait_dashboard_server.sh"]
-            )
+            ),
         ],
         name="wait_dashboard_server",
         output="screen",
+        condition=IfCondition(launch_ursim),
     )
+
     driver_starter = RegisterEventHandler(
-        OnProcessExit(target_action=wait_dashboard_server, on_exit=robot_driver)
+        OnProcessExit(target_action=wait_dashboard_server, on_exit=robot_driver),
+        condition=IfCondition(launch_ursim),
+    )
+
+    robot_driver_no_wait = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            PathJoinSubstitution(
+                [FindPackageShare("ur_robot_driver"), "launch", "ur_control.launch.py"]
+            )
+        ),
+        launch_arguments={
+            "robot_ip": robot_ip,
+            "ur_type": ur_type,
+            "launch_rviz": "false",
+            "controller_spawner_timeout": str(TIMEOUT_WAIT_SERVICE_INITIAL),
+            "initial_joint_controller": "scaled_joint_trajectory_controller",
+            "headless_mode": "true",
+            "launch_dashboard_client": "false",
+            "start_joint_controller": "false",
+            "tf_prefix": tf_prefix,
+        }.items(),
+        condition=UnlessCondition(launch_ursim),
```

Comment:
I don't particularly like having the robot driver include duplicated:
- Can you define the `robot_driver` once and then use that through a `launch.actions.GroupAction` for the non-ursim case and through `driver_starter` otherwise? That seems like a more compact way of achieving the same thing.
- More generally: Is there a reason for not wanting to wait for the dashboard server when not using ursim?

Either way, i would make sure that the driver behaves the same for both cases (in particular, that e.g. `robot_ip` is passed to both invocations).

Reply:
I will have a look at your suggestion to make it more compact, which I agree is a better idea. 

The reason for not waiting for the dashboard server and not using the IP as argument when launching the ursim is that the ip is hardcoded in the wait for dashboard server script and in the launch ursim script. But I could change it such that the configured ip is parsed to the scripts as well? However, this would require changes to the client library as well as the start usrsim scrip is located in the client library.